Bug fix: allow std 0 in the meta definition of `normal_`

nkaretnikov opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70089
* #70087
* #70086
* __->__ #70085

All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

Differential Revision: [D34089967](https://our.internmc.facebook.com/intern/diff/D34089967)

nkaretnikov committed:
Bug fix: allow std 0 in the meta definition of `normal_`

All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

[ghstack-poisoned]

nkaretnikov commented:
this one can now be merged regardless of the rest of the stack

lezcano reviewed (approved):
Nice catch!

nkaretnikov committed:
Update on "Bug fix: allow std 0 in the meta definition of `normal_`"


All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34089967).

nkaretnikov committed:
Update on "Bug fix: allow std 0 in the meta definition of `normal_`"


All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

Differential Revision: [D34089967](https://our.internmc.facebook.com/intern/diff/D34089967)

[ghstack-poisoned]

nkaretnikov committed:
Update on "Bug fix: allow std 0 in the meta definition of `normal_`"


All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

Differential Revision: [D34089967](https://our.internmc.facebook.com/intern/diff/D34089967)

[ghstack-poisoned]

Commit:
Update on "Bug fix: allow std 0 in the meta definition of `normal_`"


All other `normal` variants allow 0.  Looks like a mistake made while
copying the check.  Even the `normal_` implementation disagrees:

```
>>> t = torch.rand(2, 3, device='meta')
>>> t.normal_(mean=4, std=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: normal_ expects std > 0.0, but found std=0
>>>
>>>
>>> t = torch.rand(2, 3)
>>> t.normal_(mean=4, std=0)
tensor([[4., 4., 4.],
        [4., 4., 4.]])
```

Fixes #69523.

Differential Revision: [D34089967](https://our.internmc.facebook.com/intern/diff/D34089967)

[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34089967).
